app/controllers/authenticated_controller.rb

```diff
@@ -2,4 +2,10 @@
 
 class AuthenticatedController < ApplicationController
   include ShopifyApp::Authenticated
+
+  before_action :shop_domain
+
+  def shop_domain
+    @shop_domain = jwt_shopify_domain
+  end
```

Comment:
This is something that allows developers to implement "back to home" buttons in their multi-paged apps. It allows apps to bypass the `RequireKnownShop` concern.

Reply:
instead of `jwt_shopify_domain`, can you use `current_shopify_doman` (LoginProtection)? That helper method should have fallbacks on the `shop` param, jwt and rails session

Reply:
Fixed!